service/lib/ops/policyOps.js

```diff
@@ -39,16 +39,26 @@ function insertPolicies (client, policies, cb) {
   client.query(sql, cb)
 }
 
-function deletePolicies (client, ids, cb) {
-  client.query(SQL`DELETE FROM policies WHERE id = ANY (${ids})`, utils.boomErrorWrapper(cb))
+function deletePolicies (client, ids, orgId, cb) {
+  client.query(SQL`DELETE FROM policies WHERE id = ANY (${ids}) AND org_id=${orgId}`, utils.boomErrorWrapper(cb))
 }
 
-function deleteTeamsAssociations (client, ids, cb) {
-  client.query(SQL`DELETE FROM team_policies WHERE policy_id = ANY (${ids})`, utils.boomErrorWrapper(cb))
+function deleteTeamsAssociations (client, ids, orgId, cb) {
+  client.query(SQL`
+               DELETE FROM team_policies AS p
+               USING teams AS t
+               WHERE t.id = p.team_id
+                 AND p.policy_id = ANY (${ids})
+                 AND t.org_id = ${orgId}`, utils.boomErrorWrapper(cb))
 }
 
-function deleteUsersAssociations (client, ids, cb) {
-  client.query(SQL`DELETE FROM user_policies WHERE policy_id = ANY (${ids})`, utils.boomErrorWrapper(cb))
+function deleteUsersAssociations (client, ids, orgId, cb) {
+  client.query(SQL`
+               DELETE FROM user_policies AS p
+               USING users AS u
```

Comment:
Yes, it's the equivalent of JOIN. As far as I know you can't use JOIN in a DELETE unless you put it in a SELECT in a nested query.